src/GUI/GraphPanel.java

```diff
@@ -94,8 +92,8 @@ private Vector2D findNearestGraphPoint(Vector2D screenPosition) {
         Vector2D nearestPoint = null;
         double nearestDistance = Double.POSITIVE_INFINITY;
 
-        for (ParametricFunction function : functions) {
-            if(function == null) continue;
+        for (UUID uuid : functions.keySet()) {
```

Comment:
why not just loop through functions.values() if we only need the function, and not the ID?

Reply:
Done